### PR TITLE
Putorpatchfix

### PR DESCRIPTION
**Pull request by facchettos**
### Which issue this PR addresses:
some cleanup of putorpatch and some improvements on api registration 

Fixes

### CODE REVIEW

**Review by ross-bryan — COMMENTED** [review]
I have looked over the PR and it makes sense to me. Didn't have the time to go through it too thoroughly beyond scrolling the changes. I don't think there are any issues being introduced from this PR

File: pkg/frontend/openshiftcluster_putorpatch.go [hunk 1]
```diff
@@ -240,7 +240,7 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 	doc.OpenShiftCluster.Properties.ClusterProfile.PullSecret = ""
 	doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret = ""
 
-	b, err := json.MarshalIndent(converter.ToExternal(doc.OpenShiftCluster), "", "    ")
+	b, err := json.MarshalIndent(converter.ToExternal(doc.OpenShiftCluster), "", "\t")
```
Comment (ventifus) [inline pkg/frontend/openshiftcluster_putorpatch.go:243]: I doubt anything cares, but why change the indentation to tabs vs spaces?

**Review by ventifus — COMMENTED** [review]
lgtm

File: pkg/frontend/openshiftcluster_putorpatch.go (hunk 1 above)
Comment (facchettos) [reply pkg/frontend/openshiftcluster_putorpatch.go:243]: it's just a bit more readable imo , can revert it though, don't really care about it

**Review by bennerv — APPROVED** [review]
lgtm - makes sense was thinking the same thing when adding operations to the API `init` functions.

**Pull request merged**